### PR TITLE
[Issue 21] Support KG_BASE_URL when running kernel gateway directly

### PR DESCRIPTION
**Pull request by dalogsdon**
:chicken:

### CODE REVIEW

**Comment by dalogsdon** [comment]
part of #21

**Comment by parente** [comment]
Rebase and I'll have a look.

**Comment by dalogsdon** [comment]
This has been rebased.

**Comment by parente** [comment]
Sorry. Have to rebase once more after PR #26 went in. Since that PR also enabled system int tests, might make sense to add one for the base URL too.

**Comment by dalogsdon** [comment]
re-rebased

File: config.json [hunk 1]
```diff
@@ -1,10 +1,11 @@
 // Hjson format, see http://hjson.org for syntax
 {
-    // The URL of the kernel gateway / jupyter notebook server to contact for
-    // kernels
+    // URL of the kernel gateway / jupyter notebook server to contact for kernels
     KERNEL_GATEWAY_URL: http://192.168.99.100:8888
     // Authorization token required for kernel gateway requests
     KG_AUTH_TOKEN: null
+    // Kernel gateway base URL (gets appended to KERNEL_GATEWAY_URL)
+    KG_BASE_URL: ""
```
Comment (parente) [inline config.json]: If it's just getting appended, we probably could have just said "point kernel gateway URL **directly** to the parent path where the /api resource lives". But, OK, live and learn.

**Pull request merged**